src/org/usfirst/frc4904/robot/humaninterface/drivers/NathanGain.java

```diff
@@ -53,6 +56,21 @@ public void bindCommands() {
 		RobotMap.Component.driverXbox.dPad.right.whenPressed(new ChassisTurnAbsolute(RobotMap.Component.chassis, 90,
 			RobotMap.Component.navx, RobotMap.Component.chassisTurnMC));
 		RobotMap.Component.driverXbox.dPad.right.whenReleased(normalDrive);
+		(new Climb(new PIDSource() { // Construct a PIDSource inline so we can process and pass the axis value here
+			@Override
+			public double pidGet() {
+				double climberSpeed = -RobotMap.Component.driverXbox.rightStick.getY(); // Inverted airplane-style analog control
+				return Math.max(0, climberSpeed); // Never drive the climber at a negative speed
+			}
+
+			@Override
+			public void setPIDSourceType(PIDSourceType pidSource) {}
```

Comment:
better to `throw new UnsupportedOperationException` than to return null and cause a null pointer exception somewhere else